### PR TITLE
[BEAM-3060] Support for Perfkit execution of file-based-io-tests on HDFS cluster.

### PR DESCRIPTION
**Pull request by szewi**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by szewi** [comment]
This pull request allow to run file-based-io tests with Perfkit. It contains definition of pipeline options(pkb-config.yml) that need to be passed to beam in order to run tests. To make it working with PerfKit [#1556](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/1556) **must be merged first.** 
Pom.xml was modified that it contains path to kubernetes scripts and to pkb-config.yml. This allows to dynamically create and configure hdfs infrastructure. Based on `LoadBalancerIp` - ` --hdfsConfiguration` is created and `--filenamePrefix` is modified. 

This PR is working with DirectRunner(check note in pkb-config.yml) and with Dataflow without any additional modifications in config files by running:
`mvn verify -Dio-it-suite-hdfs-small -pl sdks/java/io/file-based-io-tests -DfileBasedIoItClass=org.apache.beam.sdk.io.text.TextIOIT -DpkbLocation="/Path_to_/PerfKitBenchmarker/pkb.py" -DintegrationTestRunner=dataflow -DpkbExtraProperties='["filesystem=hdfs"]' -DintegrationTestPipelineOptions='["--tempRoot=gs://bucket-name/"]' 
`

The jenkins configuration was provided. However additional bq tables must be created first:
- beam_performance.textioit_hdfs_pkb_results
- beam_performance.compressed_textioit_hdfs_pkb_results
- beam_performance.avroioit_hdfs_pkb_results
- beam_performance.tfrecordioit_hdfs_pkb_results

cc @chamikaramj

**Comment by chamikaramj** [comment]
Run seed job

**Comment by chamikaramj** [comment]
Run Java TextIO Performance Test HDFS

**Comment by szewi** [comment]
Run seed job

**Comment by szewi** [comment]
Run Java TextIO Performance Test HDFS

**Comment by chamikaramj** [comment]
Run Java PreCommit

**Comment by chamikaramj** [comment]
run java Gradle Precommit

**Comment by chamikaramj** [comment]
Run Seed Job

File: .test-infra/jenkins/job_beam_PerformanceTests_FileBasedIO_IT_HDFS.groovy [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+        // Allows triggering this build against pull requests.
+        common_job_properties.enablePhraseTriggeringFromPullRequest(
+                delegate,
+                testConfiguration.prCommitStatusName,
+                testConfiguration.prTriggerPhase)
+
+        // Run job in postcommit every 12 hours, don't trigger every push, and
+        // don't email individual committers.
+        common_job_properties.setPostCommit(
```
Comment (chamikaramj) [inline .test-infra/jenkins/job_beam_PerformanceTests_FileBasedIO_IT_HDFS.groovy]: We should not trigger this job till Kubernetes issues are fixed. Shall we add the test without Jenkins configs till that is figured out ?

File: sdks/java/io/file-based-io-tests/pom.xml [hunk 2]
```diff
@@ -145,6 +145,89 @@
                 </plugins>
             </build>
         </profile>
+        <profile>
+            <id>io-it-hdfs-small</id>
```
Comment (chamikaramj) [inline sdks/java/io/file-based-io-tests/pom.xml]: Please add instructions for invoking the test manually.

File: .test-infra/jenkins/job_beam_PerformanceTests_FileBasedIO_IT_HDFS.groovy (hunk 1 above)
Comment (szewi) [reply .test-infra/jenkins/job_beam_PerformanceTests_FileBasedIO_IT_HDFS.groovy]: Sure.

File: sdks/java/io/file-based-io-tests/pom.xml (hunk 2 above)
Comment (szewi) [reply sdks/java/io/file-based-io-tests/pom.xml]: ok

**Comment by chamikaramj** [comment]
run java Gradle Precommit

**Pull request merged**